### PR TITLE
Add microdox keyboard to boardsource folder

### PR DESCRIPTION
**Pull request by daysgobye**
## Description

added new keyboard to the boardsource folder as well as my map with more features enabled. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by daysgobye** [comment]
I cant tell if the ci failed because of my code or if there is problem on the ci server. If you know let me know and I can fix it.

File: keyboards/boardsource/microdox/keymaps/cole/keymap.c [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    _RAISE,
+    _LOWER,
+    _ADJUST
+};
+
+#ifdef PROTOCOL_LUFA
+#include "lufa.h"
+#include "split_util.h"
+#endif
+
```
Comment (fauxpark) [inline keyboards/boardsource/microdox/keymaps/cole/keymap.c:14]:
```suggestion
#include "split_util.h"

enum layers {
    _QWERTY,
    _RAISE,
    _LOWER,
    _ADJUST
};

```

File: keyboards/boardsource/microdox/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,40 @@
+#include QMK_KEYBOARD_H
+
+
+#ifdef PROTOCOL_LUFA
+#include "lufa.h"
+#include "split_util.h"
+#endif
+
+
```
Comment (fauxpark) [inline keyboards/boardsource/microdox/keymaps/default/keymap.c:9]:
```suggestion
#include "split_util.h"

```

File: keyboards/boardsource/microdox/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  KC_TAB,    KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT,      XXXXXXX, KC_MINS, KC_EQL,  KC_LBRC, KC_RBRC, \
+  KC_LCTL, KC_GRV,  KC_LGUI, KC_LALT, XXXXXXX,      XXXXXXX, XXXXXXX, XXXXXXX, KC_BSLS,  KC_QUOT, \
+                    XXXXXXX, XXXXXXX, XXXXXXX,      XXXXXXX, XXXXXXX, XXXXXXX                    \
+),
+
+[_LOWER] = LAYOUT( \
+  KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC,      KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, \
+  KC_ESC,  XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,      XXXXXXX, KC_UNDS, KC_PLUS, KC_LCBR, KC_RCBR, \
+  KC_CAPS, KC_TILD, XXXXXXX, XXXXXXX, XXXXXXX,      XXXXXXX, XXXXXXX, XXXXXXX, KC_PIPE,  KC_DQT, \
+                    XXXXXXX, XXXXXXX, XXXXXXX,      KC_ENT,  XXXXXXX, KC_DEL                    \
```
Comment (fauxpark) [inline keyboards/boardsource/microdox/keymaps/default/keymap.c:38]:
```suggestion
[_QWERTY] = LAYOUT(
  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,         KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,
  KC_A,    KC_S,    KC_D,    KC_F,    KC_G,         KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN,
  KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,         KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,
                    KC_LCTL, LOWER, KC_SPC,         KC_BSPC, RAISE,  KC_ENT
),

[_RAISE] = LAYOUT(
  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,         KC_6,    KC_7,    KC_8,    KC_9,    KC_0,
  KC_TAB,    KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT,      XXXXXXX, KC_MINS, KC_EQL,  KC_LBRC, KC_RBRC,
  KC_LCTL, KC_GRV,  KC_LGUI, KC_LALT, XXXXXXX,      XXXXXXX, XXXXXXX, XXXXXXX, KC_BSLS,  KC_QUOT,
                    XXXXXXX, XXXXXXX, XXXXXXX,      XXXXXXX, XXXXXXX, XXXXXXX
),

[_LOWER] = LAYOUT(
  KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC,      KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN,
  KC_ESC,  XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,      XXXXXXX, KC_UNDS, KC_PLUS, KC_LCBR, KC_RCBR,
  KC_CAPS, KC_TILD, XXXXXXX, XXXXXXX, XXXXXXX,      XXXXXXX, XXXXXXX, XXXXXXX, KC_PIPE,  KC_DQT,
                    XXXXXXX, XXXXXXX, XXXXXXX,      KC_ENT,  XXXXXXX, KC_DEL
```

File: keyboards/boardsource/microdox/rules.mk [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+AUDIO_ENABLE =        no       # Audio output on port C6
+UNICODE_ENABLE =      no       # Unicode
+BLUETOOTH_ENABLE =    no       # Enable Bluetooth
+RGBLIGHT_ENABLE =     yes      # Enable WS2812 RGB underlight.
+SPLIT_KEYBOARD =      yes
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE =    no       # Breathing sleep LED during USB suspend
+
+OLED_DRIVER_ENABLE =  no    
+RGBLIGHT_ANIMATIONS = yes
```
Comment (fauxpark) [inline keyboards/boardsource/microdox/rules.mk:25]:
```suggestion

# Bootloader selection
BOOTLOADER = caterina

# Build Options
#   change yes to no to disable
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = no        # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
SPLIT_KEYBOARD = yes
```

File: keyboards/boardsource/microdox/info.json [hunk 5]
```diff
@@ -0,0 +1,56 @@
+{
+    "keyboard_name": "microdox",
+    "url": "https://boardsource.xyz/store/5f2e7e4a2902de7151494f92",
+    "maintainer": "boardsource",
+    "width": 15,
```
Comment (noroadsleft) [inline keyboards/boardsource/microdox/info.json:5]:
```suggestion
    "width": 13,
```

File: keyboards/boardsource/microdox/info.json [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+                { "label": ".", "x": 12, "y": 2.1 },
+                { "label": "/", "x": 13, "y": 2.3 },
+
+                { "label": "GUI / KC_HANJ", "x": 4, "y": 3.7 },
+                { "label": "Lower", "x": 5, "y": 3.7 },
+                { "label": "Space", "x": 6, "y": 3.2 },
+
+                { "label": "Enter", "x": 8, "y": 3.2 },
+                { "label": "Raise", "x": 9, "y": 3.7 },
+                { "label": "Alt / KC_HAEN", "x": 10, "y": 3.7 }
```
Comment (noroadsleft) [inline keyboards/boardsource/microdox/info.json:52]:
```suggestion
                { "label": "Q", "x": 0, "y": 0.3 },
                { "label": "W", "x": 1, "y": 0.1 },
                { "label": "E", "x": 2, "y": 0 },
                { "label": "R", "x": 3, "y": 0.1 },
                { "label": "T", "x": 4, "y": 0.2 },

                { "label": "Y", "x": 8, "y": 0.2 },
                { "label": "U", "x": 9, "y": 0.1 },
                { "label": "I", "x": 10, "y": 0 },
                { "label": "O", "x": 11, "y": 0.1 },
                { "label": "P", "x": 12, "y": 0.3 },

                { "label": "A", "x": 0, "y": 1.3 },
                { "label": "S", "x": 1, "y": 1.1 },
                { "label": "D", "x": 2, "y": 1 },
                { "label": "F", "x": 3, "y": 1.1 },
                { "label": "G", "x": 4, "y": 1.2 },

                { "label": "H", "x": 8, "y": 1.2 },
                { "label": "J", "x": 9, "y": 1.1 },
                { "label": "K", "x": 10, "y": 1 },
                { "label": "L", "x": 11, "y": 1.1 },
                { "label": ";", "x": 12, "y": 1.3 },

                { "label": "Z", "x": 0, "y": 2.3 },
                { "label": "X", "x": 1, "y": 2.1 },
                { "label": "C", "x": 2, "y": 2 },
                { "label": "V", "x": 3, "y": 2.1 },
                { "label": "B", "x": 4, "y": 2.2 },

                { "label": "N", "x": 8, "y": 2.2 },
                { "label": "M", "x": 9, "y": 2.1 },
                { "label": ",", "x": 10, "y": 2 },
                { "label": ".", "x": 11, "y": 2.1 },
                { "label": "/", "x": 12, "y": 2.3 },

                { "label": "GUI / KC_HANJ", "x": 3, "y": 3.7 },
                { "label": "Lower", "x": 4, "y": 3.7 },
                { "label": "Space", "x": 5, "y": 3.2 },

                { "label": "Enter", "x": 7, "y": 3.2 },
                { "label": "Raise", "x": 8, "y": 3.7 },
                { "label": "Alt / KC_HAEN", "x": 9, "y": 3.7 }
```

File: keyboards/boardsource/microdox/info.json [hunk 7]
```diff
@@ -0,0 +1,56 @@
+{
+    "keyboard_name": "microdox",
+    "url": "https://boardsource.xyz/store/5f2e7e4a2902de7151494f92",
+    "maintainer": "boardsource",
+    "width": 15,
+    "height": 4.7,
+    "layouts": {
+        "LAYOUT": {
```
Comment (noroadsleft) [inline keyboards/boardsource/microdox/info.json:8]:
```suggestion
        "LAYOUT_split_3x5_3": {
```

File: keyboards/boardsource/microdox/microdox.h [hunk 8]
```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include "quantum.h"
+
+#define LAYOUT(\
```
Comment (noroadsleft) [inline keyboards/boardsource/microdox/microdox.h:5]:
```suggestion
#define LAYOUT_split_3x5_3(\
```

File: keyboards/boardsource/microdox/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+    _QWERTY,
+    _RAISE,
+    _LOWER,
+};
+
+#define RAISE MO(_RAISE)
+#define LOWER MO(_LOWER)
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+[_QWERTY] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/boardsource/microdox/keymaps/default/keymap.c:14]:
```suggestion
[_QWERTY] = LAYOUT_split_3x5_3(
```

File: keyboards/boardsource/microdox/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,         KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,
+  KC_A,    KC_S,    KC_D,    KC_F,    KC_G,         KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN,
+  KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,         KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,
+                    KC_LCTL, LOWER, KC_SPC,         KC_BSPC, RAISE,  KC_ENT
+),
+
+[_RAISE] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/boardsource/microdox/keymaps/default/keymap.c:21]:
```suggestion
[_RAISE] = LAYOUT_split_3x5_3(
```

File: keyboards/boardsource/microdox/keymaps/default/keymap.c [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,         KC_6,    KC_7,    KC_8,    KC_9,    KC_0,
+  KC_TAB,    KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT,      XXXXXXX, KC_MINS, KC_EQL,  KC_LBRC, KC_RBRC,
+  KC_LCTL, KC_GRV,  KC_LGUI, KC_LALT, XXXXXXX,      XXXXXXX, XXXXXXX, XXXXXXX, KC_BSLS,  KC_QUOT,
+                    XXXXXXX, XXXXXXX, XXXXXXX,      XXXXXXX, XXXXXXX, XXXXXXX
+),
+
+[_LOWER] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/boardsource/microdox/keymaps/default/keymap.c:28]:
```suggestion
[_LOWER] = LAYOUT_split_3x5_3(
```

File: keyboards/boardsource/microdox/keymaps/cole/keymap.c [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+#define SHIFT_SLASH MT(MOD_RSFT,KC_SLSH)
+#define RASE_ENT LT(_RAISE, KC_ENT)
+#define CTRL_A MT(MOD_LCTL,KC_A)
+#define PREVWINDOW MT(MOD_LCTL, KC_LEFT)
+#define NEXTWINDOW MT(MOD_LCTL, KC_RGHT)
+
+
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [_QWERTY] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/boardsource/microdox/keymaps/cole/keymap.c:24]:
```suggestion
  [_QWERTY] = LAYOUT_split_3x5_3( 
```

File: keyboards/boardsource/microdox/keymaps/cole/keymap.c [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [_QWERTY] = LAYOUT( 
+        KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,             KC_Y,    KC_U,    KC_I,    KC_O,   KC_P,  
+      CTRL_A,    KC_S,    KC_D,    KC_F,    KC_G,             KC_H,    KC_J,    KC_K,    KC_L, KC_CTSC, 
+     SHFT_Z,    KC_X,    KC_C,    KC_V,    KC_B,             KC_N,    KC_M, KC_COMM,  KC_DOT, SHIFT_SLASH, 
+                       MO(_LOWER),KC_LGUI, RASE_ENT,       RASE_BACK, LOWER_SPC, KC_TAB
+    ),
+  [_RAISE] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/boardsource/microdox/keymaps/cole/keymap.c:30]:
```suggestion
  [_RAISE] = LAYOUT_split_3x5_3( 
```

File: keyboards/boardsource/microdox/keymaps/cole/keymap.c [hunk 14]
```diff
@@ -0,0 +1,99 @@
...
+     SHFT_Z,    KC_X,    KC_C,    KC_V,    KC_B,             KC_N,    KC_M, KC_COMM,  KC_DOT, SHIFT_SLASH, 
+                       MO(_LOWER),KC_LGUI, RASE_ENT,       RASE_BACK, LOWER_SPC, KC_TAB
+    ),
+  [_RAISE] = LAYOUT( 
+        KC_1,    KC_2,    KC_3,    KC_4,    KC_5,           KC_6,    KC_7,    KC_8,    KC_9,    KC_0, 
+      KC_ESC, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,            KC_LEFT, KC_DOWN,   KC_UP, KC_RGHT, KC_QUOT, 
+     KC_LSFT, KC_GRV, PREVWINDOW, NEXTWINDOW, XXXXXXX,           KC_MINS,  KC_EQL, KC_LBRC, KC_RBRC, KC_BSLS, 
+                         _______, KC_LGUI, KC_ENT,           KC_BSPC,  KC_SPC, _______
+                         ),
+  [_LOWER] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/boardsource/microdox/keymaps/cole/keymap.c:36]:
```suggestion
  [_LOWER] = LAYOUT_split_3x5_3( 
```

File: keyboards/boardsource/microdox/config.h [hunk 15]
```diff
@@ -0,0 +1,63 @@
+#pragma once
```
Comment (noroadsleft) [inline keyboards/boardsource/microdox/config.h:1]: Need a license header here.

File: keyboards/boardsource/microdox/keymaps/cole/keymap.c [hunk 16]
```diff
@@ -0,0 +1,99 @@
+#include QMK_KEYBOARD_H
```
Comment (noroadsleft) [inline keyboards/boardsource/microdox/keymaps/cole/keymap.c:1]: Need a license header here.

File: keyboards/boardsource/microdox/keymaps/default/keymap.c [hunk 17]
```diff
@@ -0,0 +1,34 @@
+#include QMK_KEYBOARD_H
```
Comment (noroadsleft) [inline keyboards/boardsource/microdox/keymaps/default/keymap.c:1]: Need a license header here.

File: keyboards/boardsource/microdox/microdox.c [hunk 18]
```diff
@@ -0,0 +1,2 @@
+#include "microdox.h"
```
Comment (noroadsleft) [inline keyboards/boardsource/microdox/microdox.c:1]: Need a license header here.

File: keyboards/boardsource/microdox/microdox.h [hunk 19]
```diff
@@ -0,0 +1,20 @@
+#pragma once
```
Comment (noroadsleft) [inline keyboards/boardsource/microdox/microdox.h:1]: Need a license header here.

File: keyboards/boardsource/microdox/readme.md [hunk 20]
```diff
@@ -0,0 +1,13 @@
+# Microdox
+
+![boardsource Microdox](https://i.imgur.com/AliShkv.jpg)
```
Comment (noroadsleft) [inline keyboards/boardsource/microdox/readme.md:3]: Swaps this image for a smaller version.

```suggestion
![boardsource Microdox](https://i.imgur.com/AliShkvl.jpg)
```

File: keyboards/boardsource/microdox/keymaps/default/keymap.c [hunk 21]
```diff
@@ -0,0 +1,51 @@
...
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include QMK_KEYBOARD_H
+#include "split_util.h"
```
Comment (fauxpark) [inline keyboards/boardsource/microdox/keymaps/default/keymap.c:19]:
```suggestion
```
I don't think this is used here.

File: keyboards/boardsource/microdox/rules.mk [hunk 22]
```diff
@@ -0,0 +1,22 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = caterina
+
+# Build Options
+#   change yes to no to disable
+BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
```
Comment (fauxpark) [inline keyboards/boardsource/microdox/rules.mk:9]:
```suggestion
#
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
```

File: keyboards/boardsource/microdox/rules.mk [hunk 23]
```diff
@@ -7,6 +7,7 @@ BOOTLOADER = caterina
 # Build Options
 #   change yes to no to disable
 #
+#
```
Comment (fauxpark) [inline keyboards/boardsource/microdox/rules.mk:10]:
```suggestion
```

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**